src/TodoList.tsx

```diff
@@ -50,7 +50,8 @@ const TodoList = () => {
     ]);
 
     const handleClickDeleteButton = (id: number) => {
-        setTask(tasks.map(task => ({'date': task.date, 'todo': task.todo.filter(item => item.id !== id)})));
+        const newArray = tasks.map(task => ({'date': task.date, 'todo': task.todo.filter(item => item.id !== id)})).filter(item => (item.todo.length !== 0));
+        setTask(newArray);
```

Comment:
setTask의 callback에는 prevState라는 param을 제공합니다

```suggestion
        const newArray = 
        setTask((prev) => prev.map(task => ({'date': task.date, 'todo': task.todo.filter(item => item.id !== id)})).filter(item => (item.todo.length !== 0)));
```